tracker/src/main/java/org/matomo/sdk/extra/MatomoExceptionHandler.java

```diff
@@ -48,9 +48,16 @@ public Tracker getTracker() {
     public void uncaughtException(Thread thread, Throwable ex) {
         try {
             String excInfo = ex.getMessage();
-            TrackHelper.track(mTrackMe).exception(ex).description(excInfo).fatal(true).with(getTracker());
-            // Immediately dispatch as the app might be dying after rethrowing the exception
-            getTracker().dispatch();
+
+            Tracker tracker = getTracker();
+
+            TrackHelper.track(mTrackMe).exception(ex).description(excInfo).fatal(true).with(tracker);
+
+            // Force the tracker into offline mode to ensure events are written to disk
+            tracker.setOffline();
```

Comment:
This would let the Tracker attempt transmission first? While calling `setOffline` before `track` would cause no attempts to be taken, right?

Reply:
Good catch, I'll fix that.

Reply:
Fixed in 
` [cb54fc0](/matomo-org/matomo-sdk-android/pull/252/commits/cb54fc01c08ff3dda566e96538aa7c63f83f91d3)`